newsroom/notifications/utils.py

```diff
@@ -0,0 +1,104 @@
+import datetime
+
+from bson import ObjectId
+import superdesk
+
+from typing import Any
+from superdesk.utc import utcnow
+from superdesk.flask import session
+from superdesk.core import get_app_config
+from superdesk.notification import push_notification
+
+from .services import NotificationsService
+
+
+def user_notifications_lookup(user_id: str | ObjectId) -> dict[str, Any]:
+    if isinstance(user_id, str):
+        user_id = ObjectId(user_id)
+
+    ttl = get_app_config("NOTIFICATIONS_TTL", 1)
+    return {
+        "user": user_id,
+        "_created": {"$gte": utcnow() - datetime.timedelta(days=ttl)},
+    }
+
+
+async def get_user_notifications(user_id: str) -> list[dict[str, Any]]:
+    """
+    Returns the notification entries for the given user
+    """
+    lookup = user_notifications_lookup(user_id)
+    cursor = await NotificationsService().search(lookup)
+    return await cursor.to_list_raw()
+
+
+async def get_initial_notifications() -> dict[str, Any] | None:
+    """
+    Returns the stories that user has notifications for
+    :return: List of stories. None if there is not user session.
+    """
+    if not session.get("user"):
+        return None
+
+    cursor = await NotificationsService().search(lookup=user_notifications_lookup(session["user"]))
+
+    return {
+        "user": str(session["user"]) if session["user"] else None,
+        "notificationCount": await cursor.count(),
+    }
+
+
+async def get_notifications_with_items() -> dict[str, Any] | None:
+    """
+    Returns the stories that user has notifications for
+    :return: List of stories. None if there is not user session.
+    """
+    if not session.get("user"):
+        return None
+
+    saved_notifications = await get_user_notifications(session["user"])
+    item_ids = [n["item"] for n in saved_notifications]
+    items = []
+
+    try:
+        items.extend(superdesk.get_resource_service("wire_search").get_items(item_ids))
+    except (KeyError, TypeError):  # wire disabled
+        pass
+    try:
+        items.extend(superdesk.get_resource_service("agenda").get_items(item_ids))
+    except (KeyError, TypeError):  # agenda disabled
+        pass
+    try:
+        items.extend(superdesk.get_resource_service("topics").get_items(item_ids))
```

Comment:
Could this use the new async resource service to get the list of topics?

Reply:
I'm not sure about this. Isn't that what @devketanpro is working on right now?

Reply:
Sorry my bad. I was looking at my local code, which I had pulled Ketan latest from. Disregard this